src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,92 @@
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final ProductDtoMapper dtoMapper;
+
+    @PostMapping
+    @ApiOperation(value = "Save new product")
+    public ProductResponseDto create(@RequestBody ProductRequestDto requestDto) {
+        Product product = dtoMapper.mapToModel(requestDto);
+        return dtoMapper.mapToDto(productService.save(product));
+    }
+
+    @GetMapping("/{id}")
+    @ApiOperation(value = "Get product by id")
+    public ProductResponseDto getById(@PathVariable Long id) {
+        return dtoMapper.mapToDto(productService.getById(id));
+    }
+
+    @DeleteMapping("/{id}")
+    @ApiOperation(value = "Delete product by id")
+    public void deleteById(@PathVariable Long id) {
+        productService.deleteById(id);
+    }
+
+    @PutMapping
+    @ApiOperation(value = "Update product")
+    public ProductResponseDto update(@PathVariable Long id,
+                                     @RequestBody ProductRequestDto requestDto) {
+        Product product = dtoMapper.mapToModel(requestDto);
+        product.setId(id);
+        return dtoMapper.mapToDto(productService.save(product));
+    }
+
+    @GetMapping("/by-price")
+    @ApiOperation(value = "Get products with prices from - to")
+    public List<ProductResponseDto> getAllByPriceBetween(@RequestParam BigDecimal from,
+                                                         @RequestParam BigDecimal to,
+                                                         @RequestParam(defaultValue = "20")
+                                                         Integer count,
+                                                         @RequestParam(defaultValue = "0")
+                                                         Integer page,
+                                                         @RequestParam(defaultValue = "id")
+                                                         String sortBy) {
+        Sort sort = SortingPageUtil.getSort(sortBy);
+        PageRequest pageRequest = PageRequest.of(page, count, sort);
+        return productService.findAllByPriceBetween(from, to, pageRequest).stream()
+                .map(dtoMapper::mapToDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping
+    @ApiOperation(value = "Find all products")
+    public List<ProductResponseDto> findAll(@RequestParam(defaultValue = "20") Integer count,
+                                            @RequestParam(defaultValue = "0") Integer page,
```

Comment:
it's up to you but I think it's good choice to use @ApiParam in this place (some description on the front-end will be useful)